Trials.com/react-frontend/src/components/submit-run.jsx

```diff
@@ -0,0 +1,227 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Moment from "moment";
+import Star from "react-star-rating-component";
+import { useFormik, Formik } from "formik";
+import InputMask from "react-input-mask";
+import Slider from "@material-ui/core/Slider";
+import ReactStars from "react-rating-stars-component";
+
+const SubmitRun = () => {
+  const formik = useFormik({
+    initialValues: {
+      trackName: "",
+      creator: "",
+      faults: "",
+      time: "",
+      length: "",
+      faultSponginess: "",
+      ninjaLevel: "",
+      rating: "",
+    },
+    onSubmit: (values) => {
+      alert(JSON.stringify(values, null, 2));
+    },
+  });
+
+  const setRating = (ratingValue) => {
+    formik.setFieldValue("rating", ratingValue);
+  };
+
+  return (
+    <Formik>
```

Comment:
i know somehow you can set invalid inputs to appear once they click out of the input boxes. probably a few ways to do it, i know bootstrap has one, not sure if it works for formik, otherwise im sure there might just be a prop to pass for formik inputs?